### PR TITLE
cli,api: rewrite retrieving logs

### LINKED ISSUE #131: cli: follow logs, check if build exists
and if it doesn't exist, don't wait infinitely on `watch`

### LINKED ISSUE #132: logs: make it configurable to pick source of logs
* [ ] docker
* [ ] openshift

### PR DESCRIPTION
**Pull request by TomasTomecek**
new arg:

 * wait_if_missing -- wait for build to get created

new api method:

 * get_docker_build_logs -- return logs from `docker build` instead from
   build container

`BuildResponse.get_logs` is capable of decoding logs now.

Fixes #131
Fixes #132

### CODE REVIEW

**Comment by TomasTomecek** [comment]
I don't understand how test for build logs could work b/c I cannot find any code for it.

**Comment by TomasTomecek** [comment]
rebased and ready to be merged

File: osbs/cli/main.py [hunk 1]
```diff
@@ -155,12 +155,22 @@ def cmd_build_logs(args, osbs):
...
+              "Logs from docker build are part of metadata of a already built image.")
+        return
+
+    if args.from_docker_build:
+        logs = osbs.get_docker_build_logs(build_id)
     else:
-        logs = osbs.get_build_logs(build_id, follow=False, namespace=args.namespace)
-        print(logs, end="")
+        logs = osbs.get_build_logs(build_id, follow=follow,
+                                   ignore_if_missing=args.ignore_if_missing,
```
Comment (mmilata) [inline osbs/cli/main.py]: This should probably be `wait_if_missing`?

**Comment by TomasTomecek** [comment]
Very well spotted, thanks Martin! (I've changed it from `ignore-` to `wait-` and apparently forgot that one change)

**Pull request merged**